pkg/lib/packagekit.es6

```diff
@@ -264,3 +264,83 @@ export function watchRedHatSubscription(callback) {
             }
         );
     }
+
+/* Support for installing optional packages.
+ *
+ * First call check_optional_packages to determine whether something
+ * needs to be installed, then call install_optional_packages to
+ * actually install them.
+ *
+ * check_optional_packages resolves to either null when no packages
+ * need to be installed, or an object that can be passed to
+ * install_optional_packages.
+ *
+ * In the future, that object might contain more information about
+ * how many packages need to be installed, and how much space that
+ * will take, but, PackageKit...
+ */
+
+export function check_optional_packages(names, progress_cb) {
+    if (names.length == 0)
+        return cockpit.resolve(null);
```

Comment:
Please use/return standard `Promise` instead of `cockpit.defer`, as the rest of this library already does that.

Reply:
Done.